src/components/questions/question.js

```diff
@@ -1,26 +1,50 @@
 import React from "react";
 import AnswerList from "./answersList"
+import Modal from "../Modal.jsx"
+import AddAnswer from "./addAnswer.js"
 
 
 class Question extends React.Component{
   constructor(props){
     super(props);
-    this.state = {};
+    this.state = {
+      showModal: false
+    };
+    this.helpfulClick = this.helpfulClick.bind(this);
+    this.addAnswerClick = this.addAnswerClick.bind(this);
+  }
+
+  helpfulClick(event){
+    console.log("***HELPFUL CLICK***")
+  }
+
+  addAnswerClick(event){
+    this.setState({showModal:!this.state.showModal})
   }
 
   render(){
     return (<div className="container">
       <div className="container">
         <div className="title is-inline-block">{`Q: ${this.props.question_body}`}</div>
-        <div className="is-pulled-right is-inline-block">Helpful?({this.props.question_helpfulness}){" | "}Add Answer</div>
+        <div className="is-pulled-right is-inline-block">
+          <div className="is-inline-block" onClick={this.helpfulClick}>Helpful?({this.props.question_helpfulness})</div>
+          <div className="is-inline-block" onClick={this.addAnswerClick} >{" | "}Add Answer</div>
+        </div>
         <br/>
       </div>
-      {/* <div>
-      {this.props.asker_name}{" | "}{this.props.question_date}{" | "}{this.props.reported}
-      </div> */}
       <AnswerList answers={this.props.answers}/>
       <br/>
       <br/>
+      {this.state.showModal ?
+      <Modal title="Submit Your Answer"
+             children={
```

Comment:
very interesting that this works. You can also put the implemented code between the opening and closing tags of the modal if you wanted.

Reply:
i will prob make that change in the future. i forgot about that.